metrics/sinks/poseidon/driver.go

```diff
@@ -0,0 +1,114 @@
+		if isNodeMetric(metricKey, metricSet) {
+			nodeStats := stats.NodeStats{
+				Hostname:       metricSet.Labels[core.LabelHostname.Key],
+				CpuAllocatable: getMetricValue(metricSet, core.MetricNodeCpuAllocatable).FloatValue,
+				CpuCapacity:    getMetricValue(metricSet, core.MetricNodeCpuCapacity).FloatValue,
+				CpuReservation: getMetricValue(metricSet, core.MetricNodeCpuReservation).FloatValue,
+				CpuUtilization: getMetricValue(metricSet, core.MetricNodeCpuUtilization).FloatValue,
+				MemAllocatable: getMetricValue(metricSet, core.MetricNodeMemoryAllocatable).FloatValue,
+				MemCapacity:    getMetricValue(metricSet, core.MetricNodeMemoryCapacity).FloatValue,
+				MemReservation: getMetricValue(metricSet, core.MetricNodeMemoryReservation).FloatValue,
+				MemUtilization: getMetricValue(metricSet, core.MetricNodeMemoryUtilization).FloatValue,
+			}
+			client, err := sink.ReceiveNodeStats(context.Background())
+			if err != nil {
+				log.Fatalf("Error receiving node stats: %v", err)
+			}
+			client.Send(nodeStats)
+		} else if isPodMetric(metricKey, metricSet) {
+			podStats := stats.PodStats{
+				Name: metricSet.Labels[core.LabelPodName.Key],
+				Namespace: metricSet.Labels[core.LabelPodNamespace.Key],
+				Hostname: metricSet.Labels[core.LabelHostname.Key],
+				// CPU stats
+				CpuLimit: getMetricValue(metricSet, core.MetricCpuLimit).IntValue,
+				CpuRequest: getMetricValue(metricSet, core.MetricCpuRequest).IntValue,
+				CpuUsage: getMetricValue(metricSet, core.MetricCpuUsage).IntValue,
+				// Memory stats
+				MemLimit: getMetricValue(metricSet, core.MetricMemoryLimit).IntValue,
+				MemRequest: getMetricValue(metricSet, core.MetricMemoryRequest).IntValue,
+				MemUsage: getMetricValue(metricSet, core.MetricMemoryUsage).IntValue,
+				MemWorkingSet: getMetricValue(metricSet, core.MetricMemoryWorkingSet).IntValue,
+				MemPageFaults: getMetricValue(metricSet, core.MetricMemoryPageFaults).IntValue,
+				MemPageFaultsRate: getMetricValue(metricSet, core.MetricMemoryPageFaultsRate).FloatValue,
+				MajorPageFaults: getMetricValue(metricSet, core.MetricMemoryMajorPageFaults).IntValue,
+				MajorPageFaultsRate: getMetricValue(metricSet, core.MetricMemoryMajorPageFaultsRate).FloatValue,
+				//Network stats
+				NetRx: getMetricValue(metricSet, core.MetricNetworkRx).IntValue,
+				NetRxErrors: getMetricValue(metricSet, core.MetricNetworkRxErrors).IntValue,
+				NetRxErrorsRate: getMetricValue(metricSet, core.MetricNetworkRxErrorsRate).FloatValue,
+				NetRxRate: getMetricValue(metricSet, core.MetricNetworkRxRate).FloatValue,
+				NetTx: getMetricValue(metricSet, core.MetricNetworkTx).IntValue,
+				NetTxErrors: getMetricValue(metricSet, core.MetricNetworkTxErrors).IntValue,
+				NetTxErrorsRate: getMetricValue(metricSet, core.MetricNetworkTxErrorsRate).FloatValue,
+				NetTxRate: getMetricValue(metricSet, core.MetricNetworkTxRate).FloatValue,
+			}
+			client, err := sink.ReceiveNodeStats(context.Background())
```

Comment:
As above. Plus this should be sink.ReceivePodStats(context.Background(), podStats)